spec/lib/guardian_spec.rb

```diff
@@ -420,6 +420,66 @@
     end
   end
 
+  describe 'can_send_private_messages' do
+    fab!(:suspended_user) { Fabricate(:user, suspended_till: 1.week.from_now, suspended_at: 1.day.ago) }
+
+    it "returns false when the user is nil" do
+      expect(Guardian.new(nil).can_send_private_messages?).to be_falsey
+    end
+
+    it "returns false when you are untrusted" do
+      SiteSetting.personal_message_enabled_groups = Group::AUTO_GROUPS[:trust_level_2]
+      user.update!(trust_level: TrustLevel[0])
+      Group.user_trust_level_change!(user.id, TrustLevel[0])
+      expect(Guardian.new(user).can_send_private_messages?).to be_falsey
+    end
+
+    it "disallows pms to other users if trust level is not met" do
+      SiteSetting.personal_message_enabled_groups = Group::AUTO_GROUPS[:trust_level_2]
+      user.update!(trust_level: TrustLevel[1])
+      Group.user_trust_level_change!(user.id, TrustLevel[1])
+      expect(Guardian.new(user).can_send_private_messages?).to be_falsey
+    end
+
+    context "when personal_message_enabled_groups does not contain the user" do
```

Comment:
Likewise, wondering if we need a positive case for when "when personal_message_enabled_groups contains the user" with everything else 'off'